NOTE! Du måste ladda up 2020.csv, 2021.csv, 2022.csv från länken du kan hitta där nere

In [264]:
import pandas as pd
import numpy as np
from datetime import datetime, timedelta


In [265]:
EM20 = pd.read_csv('2020.csv')
EM21 = pd.read_csv('2021.csv')
EM22 = pd.read_csv('2022.csv')

print(EM20.shape[0])
print(EM21.shape[0])
print(EM22.shape[0])

print(EM20.shape[0] + EM21.shape[0] + EM22.shape[0])

11717
3115
1028
15860


# Elon Musk's Tweet data between 2010-2022



# Dataset

The dataset we're working with comes from 
https://www.kaggle.com/datasets/ayhmrba/elon-musk-tweets-2010-2021?select=2020.csv .

It contains multiple files, where the file 2020.csv contains data about Elon Musk's tweets from 2010-2020 (11718 rows)

2021.csv contains data about Elon Musk's tweets from 2021 and it contains 3115 rows

2022.csv contains data about Elon Musk's tweets from 2022 and it contains 1028 rows

In total, this data will give us 15860 tweets.

To see description about the different features, check the link.

#Differences in datasets

The 2020.csv file is a bit different from the other two files. We can see this in the example below, therefore, before we concatenate them we need to make sure that they are in the same format. 

In [71]:
print(EM20['created_at'])

print(EM22['created_at'])

0        1.609185e+12
1        1.609179e+12
2        1.609176e+12
3        1.609176e+12
4        1.609134e+12
             ...     
11712    1.322901e+12
11713    1.322900e+12
11714    1.322735e+12
11715    1.322733e+12
11716    1.275676e+12
Name: created_at, Length: 11717, dtype: float64
0       2022-03-05 12:47:21 Arabian Standard Time
1       2022-03-05 12:40:13 Arabian Standard Time
2       2022-03-05 12:32:12 Arabian Standard Time
3       2022-03-05 11:28:25 Arabian Standard Time
4       2022-03-05 09:39:29 Arabian Standard Time
                          ...                    
1023    2022-01-03 02:56:00 Arabian Standard Time
1024    2022-01-02 22:19:33 Arabian Standard Time
1025    2022-01-02 21:56:50 Arabian Standard Time
1026    2022-01-01 05:58:52 Arabian Standard Time
1027    2022-01-01 04:53:13 Arabian Standard Time
Name: created_at, Length: 1028, dtype: object


Another example we can see is that date in 2020.csv and date column in 2021 and 2022 differs.

In 2020.csv, the date column gives us both the date and time of when the tweet was uploaded. In the other two datasets it only shows us the date. 

In [5]:
print(EM20['date'])
print('-----------------------------')
print(EM21['date'])
print('-----------------------------')
print(EM22['date'])

0        2020-12-28 19:46:18
1        2020-12-28 18:07:33
2        2020-12-28 17:23:51
3        2020-12-28 17:23:31
4        2020-12-28 05:39:42
                ...         
11712    2011-12-03 08:22:07
11713    2011-12-03 08:20:28
11714    2011-12-01 10:29:04
11715    2011-12-01 09:55:11
11716    2010-06-04 18:31:57
Name: date, Length: 11717, dtype: object
-----------------------------
0       2021-12-31
1       2021-12-31
2       2021-12-31
3       2021-12-30
4       2021-12-30
           ...    
3110    2021-01-02
3111    2021-01-02
3112    2021-01-02
3113    2021-01-02
3114    2021-01-01
Name: date, Length: 3115, dtype: object
-----------------------------
0       2022-03-05
1       2022-03-05
2       2022-03-05
3       2022-03-05
4       2022-03-05
           ...    
1023    2022-01-03
1024    2022-01-02
1025    2022-01-02
1026    2022-01-01
1027    2022-01-01
Name: date, Length: 1028, dtype: object


Another Interesting part before we continue, is that the timezone is harder to interpret. In 2020.csv file, the timezone column has the same value for all rows (+000), which we interpret as the timezone +0 which is Greenwich. In the other two files, when we look at timezone we see that it uses standard Arabian time

In [6]:
print(EM20['timezone'])
print('-----------------------------')
print(EM21['timezone'])
print('-----------------------------')
print(EM22['timezone'])

0        0
1        0
2        0
3        0
4        0
        ..
11712    0
11713    0
11714    0
11715    0
11716    0
Name: timezone, Length: 11717, dtype: int64
-----------------------------
0       400
1       400
2       400
3       400
4       400
       ... 
3110    400
3111    400
3112    400
3113    400
3114    400
Name: timezone, Length: 3115, dtype: int64
-----------------------------
0       400
1       400
2       400
3       400
4       400
       ... 
1023    400
1024    400
1025    400
1026    400
1027    400
Name: timezone, Length: 1028, dtype: int64


In [7]:
#Checking tweet
print(EM21['tweet'][0])

@roshanpateI 🤣 $7


The standard arabian timezons is 4 hours ahead of the Greenwich mean time. To make sure that the hours are the same relative to eachother, we will need to readjust the times.

----------------------------------------------

# Things to fix

*  Make sure the files follow same timezone
*  Drop unnecessary columns
*  Rename the columns so that they follow the same format
*  Remove tweets that are replies/retweets

When all of these steps are done, we concatenate the dataframes and push the data to hopsworks


# This is cleaning for the 2020.csv file

Important Features needed for this task



*   Tweets
*   Amount of likes, retweets, and number of replies
*   Date and Time
*   Hashtags, cashtags, etc
*   Any media (photos, videos)


(Punkter som jag fick svar från ChatGPT)



Dropping unnecessary columns

We will need two different functions since the dataframes have minor differences that needs to be considered

In [266]:
print(EM20.columns)

Index(['Unnamed: 0', 'id', 'conversation_id', 'created_at', 'date', 'timezone',
       'place', 'tweet', 'language', 'hashtags', 'cashtags', 'user_id',
       'user_id_str', 'username', 'name', 'day', 'hour', 'link', 'urls',
       'photos', 'video', 'thumbnail', 'retweet', 'nlikes', 'nreplies',
       'nretweets', 'quote_url', 'search', 'near', 'geo', 'source',
       'user_rt_id', 'user_rt', 'retweet_id', 'reply_to', 'retweet_date',
       'translate', 'trans_src', 'trans_dest'],
      dtype='object')


In [267]:
#This drops and returns the dataframe column for 2020.csv file
def drop_cols20(df):
  #drops unnecessary columns
  drop_cols_10_20 = ['Unnamed: 0', 'id', 'conversation_id', 'place', 'hashtags', 'cashtags', 'user_id','user_id_str', 
             'link', 'quote_url', 'search', 'geo', 'near', 'source', 'translate', 'trans_src', 'trans_dest', 'retweet_date','thumbnail', 'created_at', 'user_rt_id', 'user_rt',
             'retweet_id', 'reply_to', 'hour', 'username', 'name', 'language']
  new_df = df.drop(columns=drop_cols_10_20)
  return new_df


In [268]:
newEM20 = drop_cols20(EM20)


In [269]:
newEM20 = newEM20.rename(columns={'date': 'datetime'})

In [270]:
print(newEM20.columns)

Index(['datetime', 'timezone', 'tweet', 'day', 'urls', 'photos', 'video',
       'retweet', 'nlikes', 'nreplies', 'nretweets'],
      dtype='object')


In [273]:
print(newEM20['datetime'][0])
print('-------------------------')
print(newEM20['timezone'])
print('-----------------------')
counter = 0
for i in newEM20['timezone']:
  if i == 0:
    i += 1
if counter > 0:
  print("There are multiple timezone")
else:
  print("There are only one timezone for all the times")
print('-----------------------')



2020-12-28 19:46:18
-------------------------
0        0
1        0
2        0
3        0
4        0
        ..
11712    0
11713    0
11714    0
11715    0
11716    0
Name: timezone, Length: 11717, dtype: int64
-----------------------
There are only one timezone for all the times
-----------------------


In [274]:
print(newEM20['photos'])
#either empty list or a list of an URL
print(newEM20['video'])
#Binary number if there is a video (1) or not (0)
newEM20['urls']
#same format as photos


0                                                       []
1                                                       []
2                                                       []
3                                                       []
4                                                       []
                               ...                        
11712                                                   []
11713                                                   []
11714                                                   []
11715    ['https://pbs.twimg.com/media/Afkf5DMCAAEJGFr....
11716                                                   []
Name: photos, Length: 11717, dtype: object
0        0
1        0
2        0
3        0
4        0
        ..
11712    0
11713    0
11714    0
11715    1
11716    0
Name: video, Length: 11717, dtype: int64


0                             []
1                             []
2                             []
3                             []
4                             []
                  ...           
11712                         []
11713                         []
11714    ['http://ow.ly/i/mtD7']
11715                         []
11716                         []
Name: urls, Length: 11717, dtype: object

In [275]:
#Just checking if we have any retweets, if we do we remove that tweet. but this DF does not seem to contain any retweets

counter = 0
for i in newEM20['retweet']:
  if i == True:
    i += 1
if counter > 0:
  print("There are",counter, "number of retweets")
else:
  print("There are",counter, "number of retweets")
  

There are 0 number of retweets


In [276]:
#since there are no retweets, we can drop that column
newEM20 = newEM20.drop(columns=['retweet'])


In [277]:
newEM20['day'].value_counts()
#An integer representing the days 1 = Monday, 2 = Tuesday, etc

5    1943
6    1747
4    1737
7    1701
2    1532
3    1529
1    1528
Name: day, dtype: int64

Converting Time and Date

As previously mentioned, we need to adjust the times and dates to make sure the dataframes follow the same format.

We will the create two new columns, one for the time and one for the date instead of having it as a single string in one column

In [278]:
#The value in this column will be in the same type as EM22
#print(EM22["created_at"][0])
# YYYY-MM-DD HH:MM:SS Timezone

#EM10 is in timezone 0 (greenwich) 1 hour after Swedish time
print(EM21['created_at'][0])
print('---------------------')
print(EM20['date'][0])

2021-12-31 01:11:23 Arabian Standard Time
---------------------
2020-12-28 19:46:18


In [279]:
#Function that convert the time Swedish Time
def date_time_convert(date_time, timezone):
  from datetime import datetime, timedelta
  date_time_items = date_time.split()
  time = datetime.strptime(date_time_items[1], "%H:%M:%S")

  if timezone == 0: #Greenwich timezone
    new_time = time + timedelta(hours=1)
    new_time_string = new_time.strftime("%H:%M:%S")
    date_time_items[1] = new_time_string
  elif timezone == 400: #Standard arabian timezone
    new_time = time + timedelta(hours=3)
    new_time_string = new_time.strftime("%H:%M:%S")
    date_time_items[1] = new_time_string
  
  new_date_time = " ".join(date_time_items)
  return new_date_time

#Function to call to convert all the rows in date time column
def convert_all_dates(dataframe):
  for i, v in enumerate(dataframe['date']):
    new_date_time = date_time_convert(v, dataframe['timezone'][i])
    dataframe['date'].loc[i] = new_date_time
  return dataframe

In [280]:
copy_df = newEM20.copy()

In [281]:
copy_df.head()

datetime  timezone  \
0  2020-12-28 19:46:18         0   
1  2020-12-28 18:07:33         0   
2  2020-12-28 17:23:51         0   
3  2020-12-28 17:23:31         0   
4  2020-12-28 05:39:42         0   

                                               tweet  day urls photos  video  \
0  Entertainment will be critical when cars drive...    1   []     []      0   
1  @kimpaquette Just meeting with Larry Ellison t...    1   []     []      0   
2                        @richierichhhhh_ Absolutely    1   []     []      0   
3  What should Tesla do with in-car gaming in an ...    1   []     []      0   
4                          @PPathole @WSJ Absolutely    1   []     []      0   

   nlikes  nreplies  nretweets  
0   55085      2922       2611  
1    8631       601        314  
2   72434       495        907  
3   33830      6932        884  
4    2394       131         92

In [282]:
copy_df = copy_df.rename(columns={'date': 'datetime'})

In [283]:
copy_df['datetime'] = pd.to_datetime(copy_df['datetime'])
copy_df['date'] = copy_df['datetime'].dt.date
copy_df['time'] = copy_df['datetime'].dt.time

In [284]:
copy_df.head()

datetime  timezone  \
0 2020-12-28 19:46:18         0   
1 2020-12-28 18:07:33         0   
2 2020-12-28 17:23:51         0   
3 2020-12-28 17:23:31         0   
4 2020-12-28 05:39:42         0   

                                               tweet  day urls photos  video  \
0  Entertainment will be critical when cars drive...    1   []     []      0   
1  @kimpaquette Just meeting with Larry Ellison t...    1   []     []      0   
2                        @richierichhhhh_ Absolutely    1   []     []      0   
3  What should Tesla do with in-car gaming in an ...    1   []     []      0   
4                          @PPathole @WSJ Absolutely    1   []     []      0   

   nlikes  nreplies  nretweets        date      time  
0   55085      2922       2611  2020-12-28  19:46:18  
1    8631       601        314  2020-12-28  18:07:33  
2   72434       495        907  2020-12-28  17:23:51  
3   33830      6932        884  2020-12-28  17:23:31  
4    2394       131         92  2020-12-28  05:39:42

Now we have, for each tweet, converted the time to Swedish time, aswell as created two seperate columns for time and date, we could therefore remove the datetime column

In [285]:
copy_df = copy_df.drop(columns=['datetime', 'timezone'])

In [286]:
copy_df.head()

tweet  day urls photos  video  \
0  Entertainment will be critical when cars drive...    1   []     []      0   
1  @kimpaquette Just meeting with Larry Ellison t...    1   []     []      0   
2                        @richierichhhhh_ Absolutely    1   []     []      0   
3  What should Tesla do with in-car gaming in an ...    1   []     []      0   
4                          @PPathole @WSJ Absolutely    1   []     []      0   

   nlikes  nreplies  nretweets        date      time  
0   55085      2922       2611  2020-12-28  19:46:18  
1    8631       601        314  2020-12-28  18:07:33  
2   72434       495        907  2020-12-28  17:23:51  
3   33830      6932        884  2020-12-28  17:23:31  
4    2394       131         92  2020-12-28  05:39:42

In [287]:
copy_df.shape

(11717, 10)

In [288]:
#the 2020.csv file is now cleaned and we save it as another dataframe

cleaned_2020_df = copy_df.copy()

# Cleaning for datasets 2021.csv and 2022.csv


In [289]:
EM21.columns

Index(['id', 'conversation_id', 'created_at', 'date', 'time', 'timezone',
       'user_id', 'username', 'name', 'place', 'tweet', 'language', 'mentions',
       'urls', 'photos', 'replies_count', 'retweets_count', 'likes_count',
       'hashtags', 'cashtags', 'link', 'retweet', 'quote_url', 'video',
       'thumbnail', 'near', 'geo', 'source', 'user_rt_id', 'user_rt',
       'retweet_id', 'reply_to', 'retweet_date', 'translate', 'trans_src',
       'trans_dest'],
      dtype='object')

In [290]:
EM22.columns

Index(['id', 'conversation_id', 'created_at', 'date', 'time', 'timezone',
       'user_id', 'username', 'name', 'place', 'tweet', 'language', 'mentions',
       'urls', 'photos', 'replies_count', 'retweets_count', 'likes_count',
       'hashtags', 'cashtags', 'link', 'retweet', 'quote_url', 'video',
       'thumbnail', 'near', 'geo', 'source', 'user_rt_id', 'user_rt',
       'retweet_id', 'reply_to', 'retweet_date', 'translate', 'trans_src',
       'trans_dest'],
      dtype='object')

In [291]:
counter = 0
for i in EM22['retweet']: 
  if i == True:
    i += 1
if counter > 0:
  print("There are",counter, "number of retweets")
else:
  print("There are",counter, "number of retweets")

#no retweets in EM21 and EM22, meaning we can be sure that there are no retweets in this dataset

There are 0 number of retweets


In [292]:
EM21['username']
counter = 0
for i in EM21['username']:
  if i != "elonmusk":
    counter += 1

print(counter)
#Username is only elonmusk


0


In [293]:
EM22['username']
counter = 0
for i in EM22['username']:
  if i != "elonmusk":
    counter += 1

print(counter)
#Username is only elonmusk

#Username for both datasets is unnecessary

0


In [294]:
def drop_cols21_22(df):
  #drops unnecessary columns
  drop_cols_21_22 = ['id', 'conversation_id', 'place', 'hashtags', 'cashtags', 'user_id', 
             'link', 'quote_url', 'geo', 'near', 'source', 'translate', 'trans_src', 'trans_dest', 'retweet_date','thumbnail', 'created_at', 'user_rt_id', 'user_rt',
             'retweet_id', 'reply_to', 'retweet', 'retweet_date', 'username', 'name', 'language', 'mentions']
  new_df = df.drop(columns=drop_cols_21_22)
  return new_df

In [295]:
newEM21 = drop_cols21_22(EM21)

In [296]:
newEM21.columns

Index(['date', 'time', 'timezone', 'tweet', 'urls', 'photos', 'replies_count',
       'retweets_count', 'likes_count', 'video'],
      dtype='object')

In [297]:
newEM21.head()

date      time  timezone  \
0  2021-12-31  01:11:23       400   
1  2021-12-31  00:47:53       400   
2  2021-12-31  00:28:51       400   
3  2021-12-30  22:23:14       400   
4  2021-12-30  22:15:45       400   

                                               tweet urls photos  \
0                                  @roshanpateI 🤣 $7   []     []   
1             @tesla_raj Many UI improvements coming   []     []   
2  @CSmithson80 @heydave7 @BLKMDL3 @mims This cha...   []     []   
3  @BLKMDL3 @mims Predicting macroeconomics is ch...   []     []   
4  @mims If history is any guide, not many will m...   []     []   

   replies_count  retweets_count  likes_count  video  
0            793             364        13468      0  
1           1008             435        12209      0  
2            240             143         2529      0  
3            709            1235         5756      0  
4            370             549         5247      0

Our "time" column is actually a timestamp column. So we rename it accordingly before taking only the time part of the timestamp and add them to a new column called time

In [298]:
newEM21 = newEM21.rename(columns={'time': 'timestamp'})

In [299]:
newEM21['timestamp'] = pd.to_datetime(newEM21['timestamp'], format='%H:%M:%S')
# Extract the time from each timestamp
newEM21['time'] = newEM21['timestamp'].dt.time
newEM21['time'] = newEM21['time'].astype(str)
newEM21 = newEM21.drop(columns=['timestamp'])

Then since the time is in arabian standard time (3 hours ahead of swedish time)
we need to subtract 3 hours

In [300]:
# Convert the 'date' and 'time' columns to datetime objects
newEM21['datetime'] = pd.to_datetime(newEM21['date'] + ' ' + newEM21['time'], format='%Y-%m-%d %H:%M:%S')

# Subtract 3 hours from each timestamp
newEM21['datetime'] = newEM21['datetime'] - timedelta(hours=3)

# Format the 'earlier_timestamp' column as a string
newEM21['datetime'] = newEM21['datetime'].dt.strftime('%Y-%m-%d %H:%M:%S')


In [301]:
newEM21.head()

date  timezone                                              tweet  \
0  2021-12-31       400                                  @roshanpateI 🤣 $7   
1  2021-12-31       400             @tesla_raj Many UI improvements coming   
2  2021-12-31       400  @CSmithson80 @heydave7 @BLKMDL3 @mims This cha...   
3  2021-12-30       400  @BLKMDL3 @mims Predicting macroeconomics is ch...   
4  2021-12-30       400  @mims If history is any guide, not many will m...   

  urls photos  replies_count  retweets_count  likes_count  video      time  \
0   []     []            793             364        13468      0  01:11:23   
1   []     []           1008             435        12209      0  00:47:53   
2   []     []            240             143         2529      0  00:28:51   
3   []     []            709            1235         5756      0  22:23:14   
4   []     []            370             549         5247      0  22:15:45   

              datetime  
0  2021-12-30 22:11:23  
1  2021-12-30 21:47:53  
2  2021-12-30 21:28:51  
3  2021-12-30 19:23:14  
4  2021-12-30 19:15:45

In [302]:
#Drop old times 

newEM21 = newEM21.drop(columns=['time', 'date'])


#Take the new datetime and create new time and date columns with updated time
newEM21['datetime'] = pd.to_datetime(newEM21['datetime'])
newEM21['date'] = newEM21['datetime'].dt.date
newEM21['time'] = newEM21['datetime'].dt.time


#Then we can finally drop the datetime column and timezone column as they are not needed anymore
newEM21 = newEM21.drop(columns=['datetime', 'timezone'])


Last thing we need to add to this dataset, is which day it is
(monday, tuesday, etc)

In [303]:
newEM21['date'] = pd.to_datetime(newEM21['date'], format='%Y-%m-%d')
newEM21['day']  = newEM21['date'].dt.weekday


In [305]:
newEM21.head()

tweet urls photos  \
0                                  @roshanpateI 🤣 $7   []     []   
1             @tesla_raj Many UI improvements coming   []     []   
2  @CSmithson80 @heydave7 @BLKMDL3 @mims This cha...   []     []   
3  @BLKMDL3 @mims Predicting macroeconomics is ch...   []     []   
4  @mims If history is any guide, not many will m...   []     []   

   replies_count  retweets_count  likes_count  video       date      time  day  
0            793             364        13468      0 2021-12-30  22:11:23    3  
1           1008             435        12209      0 2021-12-30  21:47:53    3  
2            240             143         2529      0 2021-12-30  21:28:51    3  
3            709            1235         5756      0 2021-12-30  19:23:14    3  
4            370             549         5247      0 2021-12-30  19:15:45    3

In [306]:
#Finally, we rename the retweets_count, likes_count, replies_count to the same name as the ones we had in 2020 fil
newEM21 = newEM21.rename(columns={'retweets_count':'nretweets', 'replies_count': 'nreplies', 'likes_count':'nlikes'})


In [307]:
cleaned_21_df = newEM21.copy()

We do the same thing for 2022.csv and then we can concatenate the files and save the whole dataset.

In [308]:
newEM22 = drop_cols21_22(EM22)

In [309]:
newEM22.head()

date      time  timezone  \
0  2022-03-05  12:47:21       400   
1  2022-03-05  12:40:13       400   
2  2022-03-05  12:32:12       400   
3  2022-03-05  11:28:25       400   
4  2022-03-05  09:39:29       400   

                                               tweet urls photos  \
0     @SpacePadreIsle In a way, this is free QA haha   []     []   
1           @teslaownersSV A beautiful home for life   []     []   
2  @SpacePadreIsle Some Starlink terminals near c...   []     []   
3                 @RationalEtienne @RogerYar Exactly   []     []   
4  @RogerYar All news sources are partially propa...   []     []   

   replies_count  retweets_count  likes_count  video  
0            250             205         4490      0  
1           1521             736        15121      0  
2            279            1022         8774      0  
3            126              96         2773      0  
4           1415            6191        43964      0

In [310]:
newEM22 = newEM22.rename(columns={'time': 'timestamp'})

In [311]:
newEM22['timestamp'] = pd.to_datetime(newEM22['timestamp'], format='%H:%M:%S')
# Extract the time from each timestamp
newEM22['time'] = newEM22['timestamp'].dt.time
newEM22['time'] = newEM22['time'].astype(str)
newEM22 = newEM22.drop(columns=['timestamp'])

In [312]:
# Convert the 'date' and 'time' columns to datetime objects
newEM22['datetime'] = pd.to_datetime(newEM22['date'] + ' ' + newEM22['time'], format='%Y-%m-%d %H:%M:%S')

# Subtract 3 hours from each timestamp
newEM22['datetime'] = newEM22['datetime'] - timedelta(hours=3)

# Format the 'earlier_timestamp' column as a string
newEM22['datetime'] = newEM22['datetime'].dt.strftime('%Y-%m-%d %H:%M:%S')

In [313]:
#Drop old times 

newEM22 = newEM22.drop(columns=['time', 'date'])


#Take the new datetime and create new time and date columns with updated time
newEM22['datetime'] = pd.to_datetime(newEM22['datetime'])
newEM22['date'] = newEM22['datetime'].dt.date
newEM22['time'] = newEM22['datetime'].dt.time


#Then we can finally drop the datetime column and timezone column as they are not needed anymore
newEM22 = newEM22.drop(columns=['datetime', 'timezone'])

In [314]:
newEM22['date'] = pd.to_datetime(newEM22['date'], format='%Y-%m-%d')
newEM22['day']  = newEM22['date'].dt.weekday

In [317]:
newEM22.head()

tweet urls photos  nreplies  \
0     @SpacePadreIsle In a way, this is free QA haha   []     []       250   
1           @teslaownersSV A beautiful home for life   []     []      1521   
2  @SpacePadreIsle Some Starlink terminals near c...   []     []       279   
3                 @RationalEtienne @RogerYar Exactly   []     []       126   
4  @RogerYar All news sources are partially propa...   []     []      1415   

   nretweets  nlikes  video       date      time  day  
0        205    4490      0 2022-03-05  09:47:21    5  
1        736   15121      0 2022-03-05  09:40:13    5  
2       1022    8774      0 2022-03-05  09:32:12    5  
3         96    2773      0 2022-03-05  08:28:25    5  
4       6191   43964      0 2022-03-05  06:39:29    5

In [316]:
#Finally, we rename the retweets_count, likes_count, replies_count to the same name as the ones we had in 2020 fil
newEM22 = newEM22.rename(columns={'retweets_count':'nretweets', 'replies_count': 'nreplies', 'likes_count':'nlikes'})

In [319]:
cleaned_22_df = newEM22.copy()

In [320]:
frames = [cleaned_2020_df, cleaned_21_df, cleaned_22_df]
concat_df = pd.concat(frames)

In [323]:
concat_df.head()


tweet  day urls photos  video  \
0  Entertainment will be critical when cars drive...    1   []     []      0   
1  @kimpaquette Just meeting with Larry Ellison t...    1   []     []      0   
2                        @richierichhhhh_ Absolutely    1   []     []      0   
3  What should Tesla do with in-car gaming in an ...    1   []     []      0   
4                          @PPathole @WSJ Absolutely    1   []     []      0   

   nlikes  nreplies  nretweets        date      time  
0   55085      2922       2611  2020-12-28  19:46:18  
1    8631       601        314  2020-12-28  18:07:33  
2   72434       495        907  2020-12-28  17:23:51  
3   33830      6932        884  2020-12-28  17:23:31  
4    2394       131         92  2020-12-28  05:39:42

In [327]:
print(concat_df.shape)

(15860, 10)


We can see that our total amount of rows is the same number as mentioned at the start of this notebook.

In [328]:
from google.colab import files

concat_df.to_csv('cleanedTweetData.csv')
files.download('cleanedTweetData.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>